### ISSUE TITLE
Updating The Voice Pack Addon While Speaking Results in Mayhem

### ISSUE THREAD

**Issue by phroggster** [EDDiscovery/EDDiscovery#1305]
Updated voice pack to V4 4.0.2.0, and right as I clicked the update button it started speaking. It clearly was not happy with that.
```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: PlaybackState has to be Stopped. Call DirectSoundOut::Stop to stop the playback.
   at CSCore.SoundOut.DirectSoundOut.Initialize(IWaveSource source)
   at AudioExtensions.AudioDriverCSCore.Start(AudioData o, Int32 vol) in D:\Users\phroggie\source\EDD\Audio\AudioDriverCSCore.cs:line 144
   at AudioExtensions.AudioQueue.Queue(AudioSample newdata, Priority p) in D:\Users\phroggie\source\EDD\Audio\AudioQueue.cs:line 185
   at AudioExtensions.AudioQueue.Submit(AudioSample s, Int32 vol, Priority p) in D:\Users\phroggie\source\EDD\Audio\AudioQueue.cs:line 265
   at ActionLanguage.ActionSay.ExecuteAction(ActionProgramRun ap) in D:\Users\phroggie\source\EDD\ActionLanguage\ActionsCoreCmds\ActionSay.cs:line 277
   at ActionLanguage.ActionRun.DoExecute() in D:\Users\phroggie\source\EDD\ActionLanguage\ActionsCore\ActionRun.cs:line 182
   at ActionLanguage.ActionRun.Execute() in D:\Users\phroggie\source\EDD\ActionLanguage\ActionsCore\ActionRun.cs:line 71
   at EDDiscovery.Actions.ActionController.ActionRun(ActionEvent ev, HistoryEntry he, ConditionVariables additionalvars, String flagstart, Boolean now) in D:\Users\phroggie\source\EDD\EDDiscovery\Actions\ActionController.cs:line 438
   at EDDiscovery.Actions.ActionController.ActionRunOnEntry(HistoryEntry he, ActionEvent ev, String flagstart, Boolean now) in D:\Users\phroggie\source\EDD\EDDiscovery\Actions\ActionController.cs:line 408
   at EDDiscovery.EDDiscoveryForm.Controller_NewEntrySecond(HistoryEntry he, HistoryList hl) in D:\Users\phroggie\source\EDD\EDDiscovery\EDDiscoveryForm.cs:line 521
   at EDDiscovery.EDDiscoveryController.NewEntry(JournalEntry je) in D:\Users\phroggie\source\EDD\EDDiscovery\EDDiscoveryController.cs:line 565
   at EliteDangerousCore.EDJournalClass.ScanTickDone(List`1 entries) in D:\Users\phroggie\source\EDD\EliteDangerous\EliteDangerous\EDJournalClass.cs:line 537
   at EliteDangerousCore.EDJournalClass.<>c__DisplayClass15_1.<ScanThreadProc>b__1() in D:\Users\phroggie\source\EDD\EliteDangerous\EliteDangerous\EDJournalClass.cs:line 507
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WndProc(Message& m)
   at ExtendedControls.SmartSysMenuForm.WndProc(Message& m) in D:\Users\phroggie\source\EDD\ExtendedControls\Forms\SmartSysMenuForm.cs:line 183
   at ExtendedControls.DraggableForm.WndProc(Message& m) in D:\Users\phroggie\source\EDD\ExtendedControls\Forms\DraggableForm.cs:line 190
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at EDDiscovery.Program.Main() in D:\Users\phroggie\source\EDD\EDDiscovery\Program.cs:line 47
```

**Comment by phroggster** [comment]
@robbyxp1 You know all about this stuff. Feel free to do with this as you will, and hopefully I'm the only one that's seen this.

**Comment by robbyxp1** [comment]
i'll have a look to see if i can do anything

**Comment by phroggster** [comment]
Is it enough to just clear the audio queues and stop any playback when *any* add-on update is requested? Seems overkill, but I don't know if it'd be possible to limit this response to only occur when updating the audio pack, since *any* add-on could also (at least in theory) be using the audio hardware. Right?

**Comment by robbyxp1** [comment]
Commit 14625b1 fixes #1305 seems to be able to swap voice packs now without crashing
